target/linux/realtek/files-5.15/drivers/net/dsa/rtl83xx/rtl930x.c

```diff
@@ -2431,13 +2431,11 @@ static void rtl930x_led_init(struct rtl838x_switch_priv *priv)
 	}
 
 	/* Set LED mode to serial (0x1) */
-	sw_w32_mask(0x3, 0x1, RTL930X_LED_GLB_CTRL);
-
-	/* Set LED active state */
-	if (of_property_read_bool(node, "active-low"))
-		sw_w32_mask(RTL930X_LED_GLB_ACTIVE_LOW, 0, RTL930X_LED_GLB_CTRL);
-	else
-		sw_w32_mask(0, RTL930X_LED_GLB_ACTIVE_LOW, RTL930X_LED_GLB_CTRL);
+	sw_w32_mask(RTL930X_LED_GLB_CTRL_LED_ACTIVE |
+	            0x3,
+	            of_property_read_bool(node, "active-high") ? RTL930X_LED_GLB_CTRL_LED_ACTIVE : 0 |
+	            0x1,
+	            RTL930X_LED_GLB_CTRL);
```

Comment:
so two points, this is just an intermediate commit, if you look at the final result, it'll all make sense, in short, I'm removing all magic values and give meaning to all values.

We can always refactor this to be slightly better too, but other then storing the intermittent value in a variable, there's not much more we can do.

As for 'still fail to see'. I still fail to see why we had to change it from `active-low` to `active-high` before? All changes are arbitrary. So before, we had it like the vendor made it. You changed it to something arbitrary that 'felt right to you'. Until the led framework 'mandates/prefers a default, we should stick (like most, I think) with whatever the vendor intended as default (how good or bad it may be), rather then arbitrary defaults.